CONTRIBUTING.md

```diff
@@ -18,6 +18,10 @@ You should at the very least run `make sanity_check` before committing anything.
 Please check you have GNU awk (gawk) installed or another UTF-8 compatible
 implementation of the AWK programming language interpreter.
 
+The script `tools/git_hook_helper.sh` helps setting up a git hook to run
+`make sanity_check` and `make bin` automatically *before* each commit.
+The option `-s` makes the hook only check for sanity (if you are in early stage
+of development).
```

Comment:
I don't understand, do you expect people to use this script more than once? (The one time being after they cloned the repository, or whenever they decide to set up the hook.)

Reply:
Not really, but some may want a hook that builds before commit (normal behaviour when doing rather limited modifications), and other might want to push commits that do not compile (in certain cases).

In short, you may want to change your hook when you're starting a big PR and you want to push content that don't compile (to your branch, so it does not involve the upstream).

Reply:
Or we can expect the user to disable the hook by him/herself if needed.

Reply:
Well, it is always a good idea to keep the code compiling, even if it is on your own branch. In general it is also a good idea to keep your history clean to make review easier, which is something that most of us do not do. So I would say that we should advise everyone to set up hooks that run sanity checks only all the time (of course there is no way to impose that), but everything should be tested by CI.

Ideally, we would like all individual commits to go through CI.

Reply:
Anyway, everything *is* tested by CI, as long as we make pull requests.

Reply:
Does it look clearer @rlepigre ?

Reply:
Yeah, that looks fine to me.